app/services/atmosphere/billing_service.rb

```diff
@@ -72,48 +84,45 @@ def self.bill_appliance(appl, billing_time, message, apply_prepayment = true)
       end
     end
 
-    # This method calculates the amount which should be billed to a specific appliance on billing_time.
+    # This method calculates the amount which should be billed to a specific deployment on billing_time.
     # It does not actually incur the charge, merely returns the appropriate information.
     # billing_time specifies when the billing action is actually performed
     # apply_prepayment determines whether the method should include prepayment period (of length specified in @appliance_prepayment_interval)
-    # - This is usually true, but will be false when calculating final charge for an appliance which is being shut down.
-    def self.calculate_charge_for_appliance(appl, billing_time, apply_prepayment = true)
-      # By default, the billing process should extend the validity of the appliance until now+1.hour.
-      billable_time = self.calculate_billable_time(appl, billing_time, apply_prepayment)
+    # - This is usually true, but will be false when calculating final charge for a deployment which is being shut down.
+    def self.calculate_charge_for_deployment(dep, billing_time, apply_prepayment = true)
+      # By default, the billing process should extend the validity of the deployment until now+1.hour.
+      billable_time = self.calculate_billable_time(dep, billing_time, apply_prepayment)
 
-      amount_due = 0
+      vm = dep.virtual_machine # Shorthand
+      # Find out how many appliances are using this VM and split costs equally
+      os_family = dep.appliance.appliance_type.os_family
+      hourly_charge = (vm.virtual_machine_flavor.get_hourly_cost_for(os_family)/vm.appliances.count).round #TODO: figure out if we should limit this to appliances with billing_state == :prepaid
+      Rails.logger.debug("Calculated hourly charge for using VM #{vm.id} is #{hourly_charge}. This VM currently has #{vm.appliances.count} appliances using it.")
+      charge = (hourly_charge*billable_time).round
 
-      # Iterate over appliance vms (there can be many, if the appliance is shared)
-      appl.virtual_machines.each do |vm|
-        # Find out how many appliances are using this VM and split costs equally
-        hourly_charge = (vm.virtual_machine_flavor.hourly_cost/vm.appliances.count).round #TODO: figure out if we should limit this to appliances with billing_state == :prepaid
-        Rails.logger.debug("Calculated hourly charge for using VM #{vm.id} is #{hourly_charge}. This VM currently has #{vm.appliances.count} appliances using it.")
-        charge = (hourly_charge*billable_time).round
-        amount_due += charge
-      end
-      # Return amount due
-      amount_due
+      # Return anticipated charge
+      charge
```

Comment:
This return statement is redundant. You can remove `charge` variable and simply return

```
(hourly_charge*billable_time).round
```